vulture_os/toolkit/api_parser/ubika/ubika.py

```diff
@@ -0,0 +1,279 @@
+        log['request']['cookies'] = new_request_cookies
+
+        context_reaction : str = log.get('context', {}).get('reaction')
+        if context_reaction:
+            log['context']['reaction'] = context_reaction.lower()
+
+        return dumps(log)
+
+    def test(self):
+        logs = []
+
+        try:
+            if (not self.ubika_access_token) or (self.ubika_access_expires_at <= (timezone.now() + timedelta(minutes=1)).timestamp()):
+                self.login()
+
+            namespace = self.ubika_namespaces[0]
+            for kind in ["security", "traffic"]:
+                if kind == "security":
+                    url = self.security_events_endpoint
+                elif kind == "traffic":
+                    url = self.traffic_logs_endpoint
+                url = url.replace("__NAMESPACE__", namespace)
+
+                logger.info(f"[{__parser__}][test]:Running tests...", extra={'frontend': str(self.frontend)})
+
+                since = timezone.now() - timedelta(minutes=15)
+                to = timezone.now() - timedelta(minutes=5)
+
+                newLogs = self.get_logs(url, since, to)
+                logs.extend(newLogs)
+
+                msg = f"{len(logs)} logs retrieved"
+                logger.info(f"[{__parser__}][test]: {msg}", extra={'frontend': str(self.frontend)})
+
+                return {
+                    "status": True,
+                    "data": logs[0:99]
+                }
+        except Exception as e:
+            logger.exception(f"[{__parser__}][test]: {e}", extra={'frontend': str(self.frontend)})
+            return {
+                "status": False,
+                "error": str(e)
+            }
+
+    def execute(self):
+        for namespace in self.ubika_namespaces:
+            if (not self.ubika_access_token) or (self.ubika_access_expires_at <= (timezone.now() + timedelta(minutes=1)).timestamp()):
+                self.login()
+
+            for kind in ["security", "traffic"]:
+                if kind == "security":
+                    url = self.security_events_endpoint
+                elif kind == "traffic":
+                    url = self.traffic_logs_endpoint
+                url = url.replace("__NAMESPACE__", namespace)
+
+                current_time = timezone.now()
+                to = None
+
+                while ((not to) or (to < current_time - timedelta(minutes=2))) and (not self.evt_stop.is_set()): # delay collect for 2 minutes
+                    since = self.last_collected_timestamps.get(f"ubika_{namespace}_{kind}") or (current_time - timedelta(hours=1))
+                    to = min(current_time - timedelta(minutes=2), since + timedelta(minutes=30)) # the more the time range is short, the less our pagination is prone to OOM
+
+                    logs = self.get_logs(url, since, to)
+
+                    logger.info(f"[{__parser__}][get_logs]: Succesfully gets {len(logs)} for url = {url}; from {since} to {to}", extra={'frontend': str(self.frontend)})
```

Comment:
```suggestion
                    logger.info(f"[{__parser__}][get_logs]: Successfully got {len(logs)} for namespace '{namespace}', for kind '{kind}', from {since} to {to}", extra={'frontend': str(self.frontend)})
```